src/components/NavBar.tsx

```diff
@@ -0,0 +1,79 @@
+import { navDesktopHeight, navMobileHeight } from '../../src/constants';
+import styled, { css } from 'styled-components';
+import theme from '../../src/theme';
+import OpenstaxLogo from './NavBarLogo';
+
+const BarWrapper = styled.div`
+  overflow: visible;
+  z-index: ${theme.zIndex.navbar};
+  background: ${theme.colors.palette.white};
+  position: relative;
+  padding: 0 ${theme.padding.navbar.mobile}rem;
+  box-shadow: 0 0.2rem 0.2rem 0 rgba(0, 0, 0, 0.1);
+  ${theme.breakpoints.desktop(css`
+    padding: 0 ${theme.padding.navbar.mobile}rem;
+  `)}
+`;
+
+const StyledNavBar = styled.div<{
+  maxWidth?: number;
+  navDesktopHeight: number;
+  navMobileHeight: number;
+}>`
+  overflow: visible;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  height: ${props => props.navMobileHeight}rem;
+  ${props => props.maxWidth ? `max-width: ${props.maxWidth}rem;` : null}
+  margin: 0 auto;
+  ${props => theme.breakpoints.desktop(css`
+    height: ${props.navDesktopHeight}rem;
+  `)}
+  @media print { display: none; }
+`;
+
+type NavBarProps = React.PropsWithChildren<{
+  maxWidth?: number;
+  navDesktopHeight?: number;
+  navMobileHeight?: number;
+}>
+
+const NavBar = (props: NavBarProps) => (
+  <BarWrapper>
+    <StyledNavBar
+      maxWidth={props.maxWidth}
+      navDesktopHeight={props.navDesktopHeight || navDesktopHeight}
+      navMobileHeight={props.navMobileHeight || navMobileHeight}
+    >
+      {props.children}
+    </StyledNavBar>
+  </BarWrapper>
+);
+
+export const NavBarWithLogo = (props: NavBarProps & {
```

Comment:
i'm not sure how clean it feels to have a separate component for the navbar with logo. a different interface might be:

```
export const PlainNavBar = () => <NavBar>NavBar</NavBar>
export const NavBarLogoAndChildren = () => <NavBar logo>Menu</NavBar>
export const NavBarLinkedLogo = () => <NavBar logo={{alt='Logo', href='/'}} />
```

where the logo prop is `boolean | React.HTMLProps<HTMLAnchorElement>}`